src/software/ai/fsm_diagram_generator.py

```diff
@@ -105,7 +105,8 @@ def remove_suffix(str, suffix):
 
     ai_dir = os.path.join(root_dir, "src/software/ai")
     for root, dirs, files in os.walk(ai_dir):
-        for file in files:
+        dirs.sort()
+        for file in sorted(files):
```

Comment:
wondering if this was done to help stop huge diffs in fsm-diagrams.md?

Reply:
yup exactly, because I noticed that every single time you ran the script, it produced a completely new output and it was because `walk` doesn't guarantee any kind of an order when you run it. forcing an order should limit the diffs to only be logical changes